src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.ActionStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final ActionStrategy actionStrategy;
+
+    public ShopServiceImpl(ActionStrategy operationStrategy) {
```

Comment:
The constructor parameter name 'operationStrategy' does not match the field name 'actionStrategy'. Consider renaming it to maintain consistency and improve readability.